src/dream.ml

```diff
@@ -64,6 +64,8 @@ include Dream__middleware.Site_prefix
 let error_template =
   Dream__http.Error_handler.customize
 
+let () = Mirage_crypto_rng_lwt.initialize ()
```

Comment:
The initialization should remain lazy. Part of Dream's design (at least outside Mirage) is that it should do as little invasive initialization as possible, unless it is actually called into. For example, Dream might be linked into some binary that has multiple sub-commands, and one command that makes the binary serve a web API for the user to locally connect a browser to. When the program is not doing Dream-related things, Dream shouldn't be feeding entropy into an RNG.

It looks like the approach from the log PR would work here &mdash; create a `ref` of what `lazy` will be forced by `random_buffer`, and set it here for the Unix/Lwt version of Dream.

Reply:
I tried to follow what I did about logs for the random initialization, hope that it's good for you. May be we should add some comments to explain the pattern and why we need such indirection (reference and lazy).

Reply:
Yes, I was already considering comments, too. However, seeing the version of the logs initialization in *this* context, it looks substantially more complicated than it needs to be. So, I suggested some simplifications, and we can write comments for the simplified version (if you agree).

Reply:
> When the program is not doing Dream-related things, Dream shouldn't be feeding entropy into an RNG.

TBH I don't really think it is worth to complicate the initialization for this niche case. If you can measure the RNG feeding, that should be very concerning for the general approach. I wouldn't bother with the Lazy, and just call `Mirage_crypto_rng_lwt.initialize ()` in the top-level of Dream.

Reply:
I would agree in principle, but just a Lazy is not that complicated, and the code in this PR could be simplified even more while keeping it (though it is already quite simple &mdash; thanks @dinosaure for making the edits!). I still would like Dream to be non-invasive whenever it is at all feasible. Apart from other considerations, it's possible (even if *extremely* unlikely) that in another part of the app, it might want to do some kind of different initialization of `Mirage_crypto_rng` for some reason. I just don't want Dream to have any unnecessary effects or co-effects when not being used.